tests/Moq.Tests/Regressions/FluentMockVisitorIssues.cs

```diff
@@ -0,0 +1,35 @@
+﻿using Xunit;
+
+namespace Moq.Tests.Regressions
+{
+	public class FluentMockVisitorIssues
+	{
+		[Fact]
+		public void WhenQueryingWithTwoIndexes_ThenSetsThemDirectly1()
+		{
+			var foo = Mock.Of<IFoo>(x => x[0].Value == "hello" && x[1].Value == "goodbye");
+
+			Assert.Equal("hello", foo[0].Value); // Fails here as foo[0] is the same object as foo[1] and foo[1].Value == "goodbye", Fixed!
+			Assert.Equal("goodbye", foo[1].Value);
+		}
+
+		[Fact]
+		public void WhenQueryingWithTwoIndexes_ThenSetsThemDirectly2()
+		{
+			var foo = Mock.Of<IFoo>(x => x[0] == Mock.Of<IBar>(b => b.Value == "hello") && x[1] == Mock.Of<IBar>(b => b.Value == "goodbye"));
+
+			Assert.Equal("hello", foo[0].Value); // These pass no problem
```

Comment:
Similar as in the above comment, there is no need to say _"these pass no problem"_ &mdash; that is what we assume for all tests, anyway.

(I know you copy & pasted these tests from the issue, and that is perfectly fine. Just pointing out that they often need to be changed a tiny bit so that they make sense in their new context.)